lib/components/Input/text-field-with-steppers/index.tsx

```diff
@@ -0,0 +1,89 @@
+    (props, ref) => {
+        const { value, onChange, disabled, textAlignment, placeholder = 0, unit = '', decimals = 2, unitPlacement = 'left' } = props;
+
+        const getFormatValue = (value: number) => {
+            let formatValue = value;
+
+            if (typeof formatValue === 'number') {
+                formatValue = parseFloat(formatValue.toFixed(decimals));
+            }
+
+            return formatValue;
+        }
+
+        const steppersSectionRight = (
+            <>
+                {textAlignment !== 'center' && (<button className={clsx("quill-input-steppers-decrement", disabled && 'quill-input-steppers--disabled')}
+                    onClick={() => {
+                        if (!disabled) {
+                            onChange?.({
+                                target: { value: getFormatValue(Number(value) - 1) },
+                            } as unknown as React.ChangeEvent<HTMLInputElement>);
+                        }
+                    }}>
+                    <LabelPairedMinusSmRegularIcon />
+                </button>)}
+                <button className={clsx("quill-input-steppers-increment", disabled && 'quill-input-steppers--disabled')}
+                    onClick={() => {
+                        if (!disabled) {
+                            onChange?.({
+                                target: { value: getFormatValue(Number(value) + 1) },
+                            } as unknown as React.ChangeEvent<HTMLInputElement>);
+                        }
+                    }}>
+                    <LabelPairedPlusSmRegularIcon />
+                </button>
+            </>
+        );
+
+        const steppersSectionLeft = (
+            <>
+                <button className={clsx("quill-input-steppers-decrement", disabled && 'quill-input-steppers--disabled')}
+                    onClick={() => {
+                        if (!disabled) {
+                            onChange?.({
+                                target: { value: getFormatValue(Number(value) - 1) },
+                            } as unknown as React.ChangeEvent<HTMLInputElement>);
+                        }
+                    }}>
+                    <LabelPairedMinusSmRegularIcon />
+                </button>
+            </>
+        );
+
+        return (
+            <div>
+                <Input
+                    {...props}
+                    ref={ref}
+                    triggerActionIcon={steppersSectionRight}
+                    leftIcon={textAlignment === 'center' && steppersSectionLeft}
+                    type="number"
+                    value={`${getFormatValue(value as number)}`}
+                    placeholder={placeholder.toLocaleString("en", { minimumFractionDigits: decimals })}
+                    leftPlaceholder={unitPlacement === "left" ? unit : undefined}
+                    rightPlaceholder={unitPlacement === "right" ? unit : undefined}
```

Comment:
```suggestion
                    leftPlaceholder={unitPlacement === "left" && unit}
                    rightPlaceholder={unitPlacement === "right" && unit}
```

Reply:
This doesn't work because in this case the type becomes false and I don't want to add boolean as a type for left and right placeholders since it's kinda meaningless

Reply:
okay I see. the type is `string`